File: problems/2225_find_players_with_zero_or_one_losses/solution.py
from typing import List


class Solution:
    def findWinners(self, matches: List[List[int]]) -> List[List[int]]:
        losses = Counter()
        players = set()
        for w, l in matches:
            players.add(w)
            players.add(l)
            losses[l] += 1
        no_loss = sorted(p for p in players if losses[p] == 0)
        one_loss = sorted(p for p in players if losses[p] == 1)
        return [no_loss, one_loss]

File: problems/2225_find_players_with_zero_or_one_losses/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    vector<vector<int>> findWinners(vector<vector<int>>& matches) {
        map<int, int> losses;
        for (auto& m : matches) {
            if (!losses.count(m[0])) losses[m[0]] = 0;
            losses[m[1]]++;
        }
        vector<int> noLoss, oneLoss;
        for (auto& [p, l] : losses) {
            if (l == 0) noLoss.push_back(p);
            else if (l == 1) oneLoss.push_back(p);
        }
        return {noLoss, oneLoss};
    }
};
